src/content/advanced/cluster-management/node-pools-capi/index.md

```diff
@@ -0,0 +1,322 @@
+You can identify the nodes' node pool using that label.
+The example `kubectl` command below will list all nodes with role, node pool name, and node name.
+
+```nohighlight
+kubectl get nodes \
+  -o=jsonpath='{range .items[*]}{.metadata.labels.kubernetes\.io/role}{"\t"}{.metadata.labels.giantswarm\.io/machine-pool}{"\t"}{.metadata.name}{"\n"}{end}' | sort
+master         ip-10-1-5-55.eu-central-1.compute.internal
+worker  a1b2c3-pool0  ip-10-1-6-225.eu-central-1.compute.internal
+worker  a1b2c3-pool0  ip-10-1-6-67.eu-central-1.compute.internal
+```
+
+{{< /tab >}}
+{{< tab id="cluster-capz-azure-vms" for-impl="capz_vms" >}}
+
+```nohighlight
+kubectl gs template cluster --provider capz --name test-cluster \
+  --organization giantswarm \
+  --description "my test cluster" \
+  --machine-pool-name pool0 \
+  --machine-pool-min-size 3 \
+  --machine-pool-max-size 5 \
+  --machine-pool-instance-type r6i.xlarge
+```
+
+The node pool name will be prepended by the cluster name. In the example above, the node pool name will be `a1b2c3-pool0`.
+
+All nodes in the node pool will be labeled with the node pool name, using the  `giantswarm.io/machine-deployment` label.
+You can identify the nodes' node pool using that label.
+The example `kubectl` command below will list all nodes with role, node pool name, and node name.
+
+```nohighlight
+kubectl get nodes \
+  -o=jsonpath='{range .items[*]}{.metadata.labels.kubernetes\.io/role}{"\t"}{.metadata.labels.giantswarm\.io/machine-deployment}{"\t"}{.metadata.name}{"\n"}{end}' | sort
+master         ip-10-1-5-55.eu-central-1.compute.internal
+worker  a1b2c3-pool0  ip-10-1-6-225.eu-central-1.compute.internal
+worker  a1b2c3-pool0  ip-10-1-6-67.eu-central-1.compute.internal
+```
+
+{{< /tab >}}
+{{< /tabs >}}
+
+## Assigning workloads to node pools {#assigning-workloads}
+
+Knowing the node pool name of the pool to use, you can use the `nodeSelector` method of assigning pods to the node pool.
+
+Assuming that the node pool name is `pool0`, and the cluster name is `a1b2c3`, your `nodeSelector` could for example look like this:
+
+{{< tabs >}}
+{{< tab id="cluster-capa-ec2" for-impl="capa_ec2" >}}
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+  nodeSelector:
+    giantswarm.io/machine-pool: a1b2c3-pool0
+```
+
+{{< /tab >}}
+{{< tab id="cluster-capz-azure-vms" for-impl="capz_vms" >}}
+
+```yaml
+A similar example for an Azure cluster:
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: nginx
+spec:
+  containers:
+  - name: nginx
+    image: nginx
+  nodeSelector:
+    giantswarm.io/machine-deployment: a1b2c3-pool0
+```
+
+{{< /tab >}}
+{{< /tabs >}}
+
+You can assign workloads to node pools in a more indirect way too. This is achieved by using other node attributes which are
+specified via the node pool and which are exposed as node labels.
```

Comment:
Can we rephrase this so it refers to automatically generated labels? It currently reads as if one needs to explicitly specify them. The given examples such as `topology.kubernetes.io/zone` are auto-generated. We don't need to explain in detail which controller does that, but maybe we can simply refer to well-known labels as listed at https://kubernetes.io/docs/reference/labels-annotations-taints/. Together with the below examples, that's enough.